### PR TITLE
Add Diary Progress Indicator plugin

### PR DESCRIPTION
**Pull request by oplosthee**
Adds diary progress indicators as can be seen here
 

I'm not entirely happy with the way the text for the diaries is formatted, but I can also not think of a better way to do it right now, so suggestions are welcome.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryProgressConfig.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+package net.runelite.client.plugins.diaryprogress;
+
+import net.runelite.client.config.Config;
+import net.runelite.client.config.ConfigGroup;
+import net.runelite.client.config.ConfigItem;
+
+@ConfigGroup(
+		keyName = "diaryprogress",
+		name = "Diary Progress",
+		description = "Configuration for the plugin for enabling and disabling diary progress indicators"
```
Comment (UniquePassive) [inline runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryProgressConfig.java]: Too many tabs (same with @ConfigItem)

File: runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryProgressPlugin.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import net.runelite.api.Varbits;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+
+@PluginDescriptor(
+		name = "Diary Progress plugin"
```
Comment (UniquePassive) [inline runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryProgressPlugin.java]: Too many tabs

File: runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryProgressPlugin.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+)
+public class DiaryProgressPlugin extends Plugin
+{
+
```
Comment (UniquePassive) [inline runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryProgressPlugin.java]: Remove newline

File: runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryEntry.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2017, oplosthee <https://github.com/oplosthee>
```
Comment (UniquePassive) [inline runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryEntry.java]: We're in 2018

File: runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryEntry.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	public String getName()
+	{
+		return name;
+	}
+
+	public Varbits[] getVarbits()
+	{
+		return varbits;
+	}
+
```
Comment (UniquePassive) [inline runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryEntry.java]: Remove newline

File: runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryEntry.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	MORYTANIA(7, "Morytania", Varbits.DIARY_MORYTANIA_EASY, Varbits.DIARY_MORYTANIA_MEDIUM, Varbits.DIARY_MORYTANIA_HARD, Varbits.DIARY_MORYTANIA_ELITE),
+	VARROCK(8, "Varrock", Varbits.DIARY_VARROCK_EASY, Varbits.DIARY_VARROCK_MEDIUM, Varbits.DIARY_VARROCK_HARD, Varbits.DIARY_VARROCK_ELITE),
+	WESTERN(10, "Western Provinces", Varbits.DIARY_WESTERN_EASY, Varbits.DIARY_WESTERN_MEDIUM, Varbits.DIARY_WESTERN_HARD, Varbits.DIARY_WESTERN_ELITE),
+	WILDERNESS(9, "Wilderness", Varbits.DIARY_WILDERNESS_EASY, Varbits.DIARY_WILDERNESS_MEDIUM, Varbits.DIARY_WILDERNESS_HARD, Varbits.DIARY_WILDERNESS_ELITE);
+
+	private final int index;
+	private final String name;
+	private final Varbits[] varbits;
+
+	DiaryEntry(int index, String name, Varbits ... varbits)
```
Comment (UniquePassive) [inline runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryEntry.java]: Varbits ... -> Varbits...

File: runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryProgressConfig.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2017, oplosthee <https://github.com/oplosthee>
```
Comment (UniquePassive) [inline runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryProgressConfig.java]: We're in 2018

File: runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryProgressPlugin.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2017, oplosthee <https://github.com/oplosthee>
```
Comment (UniquePassive) [inline runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryProgressPlugin.java]: We're in 2018

File: runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryProgressConfig.java (hunk 7 above)
Comment (Hual) [reply runelite-client/src/main/java/net/runelite/client/plugins/diaryprogress/DiaryProgressConfig.java]: You're writing runescape-related content and saying we're in 2018

**Comment by oplosthee** [comment]
Requested changes have been made and previous commit has been amended.

**Pull request merged**